### PR TITLE
Feature/filestorage sharenetworks update

### PR DESCRIPTION
**Pull request by ehdou**
Add support for Share Network Update
For #130

### CODE REVIEW

**Comment by ehdou** [comment]
[Update function](https://github.com/openstack/manila/blob/16c522bf5e5c3d37018caef11dd7852462193327/manila/api/v1/share_networks.py#L205) from the Manila code.

**Comment by ehdou** [comment]
@jrperritt This PR is ready for review. It's based on [PR#125](https://github.com/gophercloud/gophercloud/pull/125) and add support for update

**Comment by mvalkon** [comment]
@ehdou please replace #114 with #130 in the PR description.

**Comment by ehdou** [comment]
@jrperritt This PR has been rebased and is ready for review

File: openstack/sharedfilesystems/v2/sharenetworks/testing/requests_test.go [hunk 1]
```diff
@@ -168,3 +168,16 @@ func TestGet(t *testing.T) {
...
+
+// Verifies that it is possible to update a share network
+func TestUpdate(t *testing.T) {
+	th.SetupHTTP()
+	defer th.TeardownHTTP()
+
+	MockUpdateResponse(t)
+
+	options := sharenetworks.UpdateOpts{Name: "net_my2"}
+	v, err := sharenetworks.Update(client.ServiceClient(), "713df749-aac0-4a54-af52-10f6c991e80c", options).Extract()
```
Comment (jrperritt) [inline openstack/sharedfilesystems/v2/sharenetworks/testing/requests_test.go]: This should be more thorough: all the `Update` fields should be tested. Also, the full `ShareNetwork` should be `Extract`ed and tested with `th.AssertDeepEquals`.

Comment (ehdou) [inline openstack/sharedfilesystems/v2/sharenetworks/testing/requests_test.go]: Sure that makes sense, fixed

**Comment by jrperritt** [comment]
+2

**Pull request merged**